### DISCUSSION TITLE
RFC: Add a task to create an app based on user input

### DISCUSSION THREAD

**Discussion by onedr0p** [onedr0p/cluster-template#1139] [Ideas]
With using the `app-template` there could be a task that takes user input and generates the YAML for it's deployment, this would only handle "simple" applications where anything below can be used

It could ask simple questions like:

1. Application Namespace?
1. Application Name?
1. Application Container with Tag?
1. Application Env?
1. Application Args?
1. Application Port?
1. Application Storage Class?
1. Application Persistence Size?

I am generally looking for feedback on any of this.

**Comment by MegaShinySnivy** [comment]
I think it would be quite useful, would certainly cut out a lot of 'boilerplate' writing that is shared between most apps. I would also want to add some sort of volsync integration, not only for automated configuration of backups, but also for those migrating from V1 to V2 app template or similar.

**Reply by MegaShinySnivy** [reply]
Update to this: ingresses and annotations for them would be nice

**Reply by onedr0p** [reply]
Not sure where I stand with this because it's a ton of inputs to ask the user and I need to maintain the template over the template which can have underlaying templates, this is why I converted it to a discussion. 😆
